### ISSUE TITLE
Inference on a trained model.

### ISSUE THREAD

**Issue by bibs2091** [yvquanli/TrimNet#5]
Hello,
I have trained the TrimNet on the `clintox` dataset on the FDA_APPROVED task and it is not very obvious how to inference on new string data, can you show how to infer the model on a single SMILE string?
Thank you

**Comment by yvquanli** [comment]
If you have trained your model, you will have an triner(instance) of Trainer (class) in trainer.py.

Then, try this, I write this in text model, so, there may have bugs in these lines of code


`
    
    from dataset import atom_attr, bond_attr
    mol = Chem.MolFromSMILES(smiles)
    node_attr = atom_attr(mol)
    edge_index, edge_attr = bond_attr(mol)
    data = Data(
            x=torch.FloatTensor(node_attr),
            # pos=pos,
            edge_index=torch.LongTensor(edge_index).t(),
            edge_attr=torch.FloatTensor(edge_attr),
            y=None  # None as a placeholder
      )
    from torch_geometric.data import Batch
    batch = Batch().from_data_list([data])
    out = trainer.model(batch)

`

**Comment by bibs2091** [comment]
@yvquanli Thank you so much. 
One question though: why is the model's output is a size of **2** ? I have only one task `FDA_APPROVED`, shouldn't it be a binary classification? I thought the last layer is a softmax but apparently, the output doesn't sum up to 1 (I checked your paper quickly and I didn't find a clue to the answer)

**Comment by yvquanli** [comment]
This is because the output actually is the logits you can change it to the final prediction by

    probs=torch.softmax(logits)
    pred=torch.max(probs)

**Comment by bibs2091** [comment]
Thank you, this is helpful!
